src/typeahead/typeahead.spec.ts

```diff
@@ -0,0 +1,211 @@
+
+           // clicking not selected
+           fixture.componentInstance.model = 'o';
+           fixture.detectChanges();
+           expectWindowResults(compiled, ['+one', 'one more']);
+           expect(getNativeInput(compiled).value).toBe('o');
+
+           getWindowLinks(fixture.debugElement)[0].triggerEventHandler('click', {});
+           fixture.detectChanges();
+           expect(getWindow(compiled)).toBeNull();
+           expect(getNativeInput(compiled).value).toBe('one');
+         });
+       })));
+
+    it('should select the result on ENTER, close window and fill the input',
+       async(inject([TestComponentBuilder], (tcb) => {
+         const html = `<input type="text" [(ngModel)]="model" [ngbTypeahead]="find"/>`;
+
+         tcb.overrideTemplate(TestComponent, html).createAsync(TestComponent).then((fixture) => {
+           fixture.detectChanges();
+           const compiled = fixture.nativeElement;
+
+           fixture.componentInstance.model = 'o';
+           fixture.detectChanges();
+           expectWindowResults(compiled, ['+one', 'one more']);
+           expect(getNativeInput(compiled).value).toBe('o');
+
+           getDebugInput(fixture.debugElement).triggerEventHandler('keyup.enter', {});
+           fixture.detectChanges();
+           expect(getWindow(compiled)).toBeNull();
+           expect(getNativeInput(compiled).value).toBe('one');
+           expect(fixture.componentInstance.model).toBe('one');
+         });
+       })));
+
+    it('should select previous/next results with up/down arrow keys', async(inject([TestComponentBuilder], (tcb) => {
+         const html = `<input type="text" [(ngModel)]="model" [ngbTypeahead]="find"/>`;
+
+         tcb.overrideTemplate(TestComponent, html).createAsync(TestComponent).then((fixture) => {
+           fixture.detectChanges();
+           const compiled = fixture.nativeElement;
+
+           fixture.componentInstance.model = 'o';
+           fixture.detectChanges();
+           expectWindowResults(compiled, ['+one', 'one more']);
+
+           // down
+           getDebugInput(fixture.debugElement).triggerEventHandler('keyup.arrowDown', {});
+           fixture.detectChanges();
+           expectWindowResults(compiled, ['one', '+one more']);
+
+           getDebugInput(fixture.debugElement).triggerEventHandler('keyup.arrowDown', {});
+           fixture.detectChanges();
+           expectWindowResults(compiled, ['+one', 'one more']);
+
+           // up
+           getDebugInput(fixture.debugElement).triggerEventHandler('keyup.arrowUp', {});
+           fixture.detectChanges();
+           expectWindowResults(compiled, ['one', '+one more']);
+
+           getDebugInput(fixture.debugElement).triggerEventHandler('keyup.arrowUp', {});
+           fixture.detectChanges();
+           expectWindowResults(compiled, ['+one', 'one more']);
+         });
+       })));
+
+  });
+
+  xdescribe('conversion', () => {
```

Comment:
`gulp lint` is fine, no worries. Will push a couple of commits to fix the build, your comments and `keyup` after you've finished